lib/mllp/client.ex

```diff
@@ -464,6 +468,34 @@ defmodule MLLP.Client do
     stop_connection(state, reason, "process terminated")
   end
 
+  defp recv_ack(state, opts, buffer) do
```

Comment:
There's one missing piece for this and that is a timeout or number of times we will attempt to pull chunks off the socket before returning an error.

Reply:
Note there are at least two ways to handle this. Modify the function `recv_ack` to decrement the timeout for each pass.  Other option is to spin up a task that calls this function, and recv merely uses a really long timeout.

The latter is unappealing because we might block the socket for way too long. 

This also leads to a problem we currently have, we allow for `:infinity` as the reply timeout, which is just generally problematic.  If infinity is allowed, we can block on the backend or on the socket, well for infinity.

Reply:
I pushed up an initial version of the former (timeout based loop)

Reply:
Alas (and sorry for the noise), there may be a third option, that is specify the line delimiter (in this case it would be the trailer), I believe this can only be used an active mode, let me experiment...